apps/protocol-frontend/src/components/ContributionsByDateChart.tsx

```diff
@@ -0,0 +1,200 @@
+    return {
+      day: contribution.date,
+      value: contribution.count,
+    };
+  });
+
+  if (isError) {
+    return (
+      <Text>An error occurred fetching the DAO's recent Contributions.</Text>
+    );
+  }
+
+  if (isFetching || isLoading) {
+    return <GovrnSpinner />;
+  }
+
+  const CustomTooltipDailyContributions = ({
+    day,
+    value,
+  }: CalendarTooltipProps) => {
+    const { data: dailyContributions } = useContributionList({
+      first: 1000, // TODO: better way to do this -- do we want to have to use infinite list for this? we can also cap it at 10 or something
+      where: {
+        guilds: { some: { guild: { is: { id: { equals: daoId } } } } },
+        date_of_engagement: {
+          gte: new Date(day),
+          lt: addDays(new Date(day), 1),
+        },
+      },
+    });
+
+    return (
+      <TooltipWrapper anchor="left" position={[0, 0]}>
+        <Flex
+          direction="column"
+          justifyContent="center"
+          alignItems="flex-start"
+          width={{ base: '75vw', lg: '25vw' }}
+          backgroundColor="white"
+          boxShadow="md"
+          padding={2}
+        >
+          <Text fontWeight="normal" fontSize="sm">
+            {value} {pluralize('Contributions', parseInt(value))} on {day}
+          </Text>
+          <Divider marginY={1} />
+          {dailyContributions?.map(contribution => (
+            <Text as="span" fontWeight="normal" fontSize="xs">
+              {contribution.name} -
+              <Text
+                as="span"
+                fontWeight="normal"
+                fontSize="xs"
+                bgGradient="linear(to-l, #7928CA, #FF0080)"
+                bgClip="text"
+              >
+                {' '}
+                {contribution.user.name}
+              </Text>
+            </Text>
+          ))}
+        </Flex>
+      </TooltipWrapper>
+    );
+  };
+
+  return (
+    <Flex direction="column" paddingBottom={4} paddingX={{ base: 4, lg: 0 }}>
+      <Flex
+        direction="column"
+        paddingY={{ base: '4', lg: '4' }}
+        paddingX={{ base: '0', lg: '4' }}
+        color="gray.700"
+        background="white"
+        boxShadow="sm"
+        borderRadius={{ base: 'none', md: 'md' }}
+      >
+        <Heading
+          as="h3"
+          size="md"
+          color="gray.800"
+          fontWeight="normal"
+          paddingBottom={4}
+          textAlign="center"
+        >
+          Contributions By Date
+        </Heading>
+        {contributionsCountMap?.length !== 0 && (
+          <Box
+            alignSelf="flex-start"
+            height={{ base: '5rem', lg: '10rem' }}
+            width="100%"
+            paddingX={{ base: 2, lg: 0 }}
+          >
+            <ResponsiveTimeRange
+              data={
+                contributionsCountMap
+                  ? contributionsCountMap.filter(
+                      contribution => contribution.value !== 0,
+                    )
+                  : []
```

Comment:
Is this check necessary with the check above?

Reply:
I was talking more about the turnery expression. Doesn't line 253 already do this check. We can have the filter without the turnery.